src/Elmish.WPF/ViewModel.fs

```diff
@@ -582,13 +638,13 @@ and [<AllowNullLiteral>] internal ViewModel<'model, 'msg>
         false
     | true, binding ->
         match binding with
-        | TwoWay (_, set, dispatch)
-        | TwoWayValidate (_, set, _, dispatch) ->
+        | TwoWay { Set = set; Dispatch = dispatch }
+        | TwoWayValidate { Set = set; Dispatch = dispatch } ->
             set value currentModel |> dispatch
             true
-        | SubModelSelectedItem (_, _, set, name, dispatch) ->
+        | SubModelSelectedItem { Set = set; SubModelSeqBindingName = name; Dispatch = dispatch } ->
```

Comment:
I'm thinking it might be less verbose/noisy to not destructure like this, but instead consistently (as long as it makes sense) bind the record to a value, throughout the file. What do you think?

Reply:
Yea, I know what you mean.  Normally I would have done that in this case, but `SubModelSelectedItem` is going to change a lot in the next PR, so I decided to go with the deconstruction here for now since it is easier to see that behavior was preserved.

I think deconstruction is required when cases are combined.  I also deconstructed if only one or two fields were used.  I think of those as the special cases.  In general, I agree with binding the record to an identifier.

Are there any specific deconstructions (other than for `SubModelSelectedItem`) that would like me to remove?

Reply:
Personally I'd like all of these record deconstructions to just use e.g. `b` (because `Binding` suffix) and dot notation, but I won't force the issue if you don't like that.

Reply:
One nice thing about deconstructing is that searching on the DU case name in Visual Studio gives more information.  In that penultimate result, I know that only the `Values` field is used.  That helps me in my static analysis.

![Screenshot 2019-10-01 08 09 24](https://user-images.githubusercontent.com/34664007/65965246-ea747600-e423-11e9-9e00-8b48120e6c7a.png)

Reply:
Suit yourself :) 👍